core/domain/service.go

```diff
@@ -182,9 +182,30 @@ func (s Service) ListOrgByDomain(ctx context.Context, email string) ([]string, e
 		return nil, err
 	}
 
+	// check if user is already a member of the organization. if yes, do not include the org in the response
```

Comment:
We should rename the method name or accept a filter argument to only list orgs the user is not part of. Current implementation `ListOrgByDomain` doesn't really list all orgs by domain.